### PR TITLE
QA: Add CentOS 7 migration test

### PR DESCRIPTION
**Pull request by nodeg**
## What does this PR change? 

This will add a test for migrating a CentOS 7 traditional client to a Salt minion.

## GUI diff

No difference.

- [x] **DONE**

## Documentation

- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage

- Cucumber tests were added

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/6782
[Manager 4.2](https://github.com/SUSE/spacewalk/pull/17031)
[Manager 4.1](https://github.com/SUSE/spacewalk/pull/17032)

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by nodeg** [comment]
At the moment I got the following error when running the tests:

```bash
(...)
Tansaction Summary
================================================================================
Install  6 Packages (+26 Dependent packages)
Upgrade             (  3 Dependent packages)

Total download size: 7.9 M
Downloading packages:
Delta RPMs disabled because /usr/bin/applydeltarpm not installed.
http://download.opensuse.org/repositories/systemsmanagement%3A/Uyuni%3A/Master%3A/CentOS7-Uyuni-Client-Tools/CentOS_7/noarch/mgr-osad-4.3.3-2.3.uyuni.noarch.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
To address this issue please refer to the below wiki article 

https://wiki.centos.org/yum-errors

If above article doesn't help to resolve this issue please use https://bugs.centos.org/.

http://download.opensuse.org/repositories/systemsmanagement%3A/Uyuni%3A/Master%3A/CentOS7-Uyuni-Client-Tools/CentOS_7/noarch/python2-mgr-osa-common-4.3.3-2.3.uyuni.noarch.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://download.opensuse.org/repositories/systemsmanagement%3A/Uyuni%3A/Master%3A/CentOS7-Uyuni-Client-Tools/CentOS_7/noarch/python2-mgr-osad-4.3.3-2.3.uyuni.noarch.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://download.opensuse.org/repositories/systemsmanagement%3A/Uyuni%3A/Master%3A/CentOS7-Uyuni-Client-Tools/CentOS_7/noarch/python2-spacewalk-check-4.3.5-4.1.uyuni.noarch.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://download.opensuse.org/repositories/systemsmanagement%3A/Uyuni%3A/Master%3A/CentOS7-Uyuni-Client-Tools/CentOS_7/noarch/python2-spacewalk-client-setup-4.3.5-4.1.uyuni.noarch.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://download.opensuse.org/repositories/systemsmanagement%3A/Uyuni%3A/Master%3A/CentOS7-Uyuni-Client-Tools/CentOS_7/noarch/python2-spacewalk-client-tools-4.3.5-4.1.uyuni.noarch.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://download.opensuse.org/repositories/systemsmanagement%3A/Uyuni%3A/Master%3A/CentOS7-Uyuni-Client-Tools/CentOS_7/noarch/spacewalk-check-4.3.5-4.1.uyuni.noarch.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://download.opensuse.org/repositories/systemsmanagement%3A/Uyuni%3A/Master%3A/CentOS7-Uyuni-Client-Tools/CentOS_7/noarch/spacewalk-client-setup-4.3.5-4.1.uyuni.noarch.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://download.opensuse.org/repositories/systemsmanagement%3A/Uyuni%3A/Master%3A/CentOS7-Uyuni-Client-Tools/CentOS_7/noarch/spacewalk-client-tools-4.3.5-4.1.uyuni.noarch.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.


Error downloading packages:
  python2-mgr-osa-common-4.3.3-2.3.uyuni.noarch: [Errno 256] No more mirrors to try.
  python2-spacewalk-check-4.3.5-4.1.uyuni.noarch: [Errno 256] No more mirrors to try.
  python2-spacewalk-client-setup-4.3.5-4.1.uyuni.noarch: [Errno 256] No more mirrors to try.
  python2-mgr-osad-4.3.3-2.3.uyuni.noarch: [Errno 256] No more mirrors to try.
  spacewalk-check-4.3.5-4.1.uyuni.noarch: [Errno 256] No more mirrors to try.
  python2-spacewalk-client-tools-4.3.5-4.1.uyuni.noarch: [Errno 256] No more mirrors to try.
  spacewalk-client-tools-4.3.5-4.1.uyuni.noarch: [Errno 256] No more mirrors to try.
  spacewalk-client-setup-4.3.5-4.1.uyuni.noarch: [Errno 256] No more mirrors to try.
  mgr-osad-4.3.3-2.3.uyuni.noarch: [Errno 256] No more mirrors to try.
```

**Comment by nodeg** [comment]
@srbarrios In the issue you mentioned
> When done, remove from galaxy/testplan/manual-cucumber.md

but I do not see where I should do this exactly.

**Comment by srbarrios** [comment]
> @srbarrios In the issue you mentioned
> 
> > When done, remove from galaxy/testplan/manual-cucumber.md
> 
> but I do not see where I should do this exactly.

Actually this card was created long time ago by another squad member that left, I'm not sure where she refers, but I tried to find it by myself and I also did not found it here https://gitlab.suse.de/galaxy/testplan/-/blob/master/manual-cucumber.md, the closest section we have is https://gitlab.suse.de/galaxy/testplan/-/blob/master/manual-cucumber.md#migrationsteps

File: testsuite/features/step_definitions/command_steps.rb [hunk 1]
```diff
@@ -1420,6 +1420,16 @@
...
 end
 
+When(/^I refresh the packages list via package manager on "([^"]*)"$/) do |host|
+  node = get_target(host)
+  if host.include? 'ceos'
+    cmd_clean = "yum -y clean all"
+    cmd_cache = "yum -y makecache"
+  end
+  node.run(cmd_clean)
+  node.run(cmd_cache)
```
Comment (srbarrios) [inline testsuite/features/step_definitions/command_steps.rb:1430]:
```suggestion
  next unless host.include? 'ceos'
    
  node.run('yum -y clean all')
  node.run('yum -y makecache')
```

**Comment by nodeg** [comment]
The scenario titles should be unique because I added a context at the end.

File: testsuite/features/step_definitions/command_steps.rb (hunk 1 above)
Comment (Bischoff) [reply testsuite/features/step_definitions/command_steps.rb:1430]: Also, is this step needed at all ? Isn't `yum` doing a refresh automatically at each operation?

Comment (srbarrios) [reply testsuite/features/step_definitions/command_steps.rb:1430]: If so, if we want this step to be only used for RH-compatible hosts, we should somehow clarify the step definition text.

For instance:
`When(/^I refresh the packages list via YUM package manager on "([^"]*)"$/) do |host|`

**Comment by nodeg** [comment]
@Bischoff Apparently not. If I do not include this step I ran into the issue shown in my second comment. I explicitly need to refresh the the repositories in order to get the right package (build)versions.

File: testsuite/features/step_definitions/command_steps.rb (hunk 1 above)
Comment (nodeg) [reply testsuite/features/step_definitions/command_steps.rb:1430]: I wanted to leave it unspecific for future features. Maybe we want to include Debian/Ubuntu hosts for some reasons as well.

**Comment by srbarrios** [comment]
> @Bischoff Apparently not. If I do not include this step I ran into the issue shown in my second comment. I explicitly need to refresh the the repositories in order to get the right package (build)versions. Yes this step is (at the moment) exclusively for RHEL compatible hosts.

I think this is linked to the repository definition, you need to have `autorefresh=1` inside the .repo file, in order to force this refresh automatically by the yum command, otherwise, you need to trigger it manually as you do here.

**Comment by nodeg** [comment]
I was not able to find anything similar in the `yum.conf` [documentation](https://linux.die.net/man/5/yum.conf).

**Comment by Bischoff** [comment]
Thanks for the clarification @nodeg and @srbarrios .

**Review by Bischoff — APPROVED** [review]

**Pull request merged**